lib/VolSurface/Calibration/Equities.pm

```diff
@@ -0,0 +1,637 @@
+    my $atm_alpha2 = $params->{atmWingR};
+
+    my $atm_vol =
+        $sig_inf +
+        ($sig_1y - $sig_inf) *
+        ((exp(-$atm_alpha2 * $tiy) - exp(-$atm_alpha1 * $tiy)) / (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small))) +
+        ($sig_small - $sig_inf) *
+        ((exp(-$atm_alpha2 * $t_small - $atm_alpha1 * $tiy) - exp(-$atm_alpha1 * $t_small - $atm_alpha2 * $tiy)) /
+            (exp(-$atm_alpha2 * $t_small) - exp(-$atm_alpha1 * $t_small)));
+
+    return $atm_vol;
+}
+
+# Algorithm change - now based on centroid calculations
+# A function that optimizes a set of parameters against a function.
+# This optimization method is based on Amoeba optimization
+
+sub _optimize {
+    my ($self, $params) = @_;
+
+    my $intol         = $self->_tolerance_level;
+    my $num_of_var    = scalar(@$params);
+    my $num_of_points = $num_of_var;
+
+    my $highest_value;
+
+    my $lambda1 = ((($num_of_points + 1)**0.5) - 1 + $num_of_points) / ($num_of_points * (2**0.5)) * $self->_scale;
+    my $lambda2 = ((($num_of_points + 1)**0.5) - 1) / ($num_of_points * (2**0.5)) * $self->_scale;
+
+    my $i;
+    my $j;
+
+    #initialize the simplex with the initial guess
+    my @simplex = ($params);
+
+    my $step = 0;
+    for ($i = 0; $i < $num_of_var; $i++) {
+        $step = 0.1 * abs($params->[$i])
+            if (0.1 * abs($params->[$i]) > $step);
+    }
+
+    for (my $i = 1; $i <= $num_of_points; $i++) {
+        my @new = map { ($i - 1 == $_) ? $simplex[0][$_] + $step : 0 + $simplex[0][$_] } (0 .. $num_of_var - 1);
+        push @simplex, \@new;
+    }
+
+    my @function_eval = map { $self->function_to_optimize($_) } @simplex;
+    my $starting_value = $function_eval[0];
+
+    my @current_reflection;
+    my @current_expansion;
+    my @current_contraction;
+
+    my $expansion_function_eval;
+    my $reflected_function_eval;
+    my $contraction_function_eval;
+    my $lowest_point_index = 0;
+
+    my $calcvert = 1;
+    my $counter  = 0;
+
+    while (1) {
+        if ($calcvert == 1) {
+            $calcvert = 0;
+            for ($j = 0; $j <= $num_of_points; $j++) {
+                if ($j != $lowest_point_index) {
+                    for ($i = 0; $i < $num_of_points; $i++) {
+                        $current_reflection[$i] = $simplex[$j][$i];
+                    }
+                    $function_eval[$j] = $self->function_to_optimize(\@current_reflection);
+                }
+            }
+
+        }
+        my @sorted_eval = sort { $a <=> $b } @function_eval;
+        $lowest_point_index = first_index { $_ == $sorted_eval[0] } @function_eval;
+        my $highest_point_index = first_index { $_ == $sorted_eval[-1] } @function_eval;
+        my @indexes = indexes { $_ == $sorted_eval[-2] } @function_eval;
+
+        my @simplex_centroid = _calculate_simplex_centroid(\@simplex, $num_of_points, $num_of_var, $highest_point_index);
+
+        $highest_value = $function_eval[$highest_point_index];
```

Comment:
Just for code cleanliness, the lowest value is used below at least as often as the highest value. What makes the highest value so special that it deserves a separate variable while the lowest does not?

If I were you, I would drop this variable instead of introducing `$lowest_value`.